gcloud/pubsub/client.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from gcloud._helpers import get_default_project
+from gcloud._helpers import _ClientProxy
+from gcloud.pubsub._implicit_environ import _require_connection
+from gcloud.pubsub import api
+from gcloud.pubsub.subscription import Subscription
+from gcloud.pubsub.topic import Topic
+
+
+class Client(object):
+    """Wrap :mod:`gcloud.pubsub` API objects.
+
+    :type connection: :class:`gcloud.pubsub.connection.Connection` or None
+    :param connection: The configured connection. Defaults to one inferred
+                       from the environment.
+
+    :type project: str or None
+    :param connection: The configured project. Defaults to the value inferred
+                       from the environment.
+    """
+
+    def __init__(self, connection=None, project=None):
+        self.connection = _require_connection(connection)
+        if project is None:
+            project = get_default_project()
+        self.project = project
+
+    def topic(self, name):
```

Comment:
> Why? Nothing prevents you from doing this.

Sure... and nothing prevents you from wasting effort by putting an expensive method call inside a loop...

```python
for i in range(0, 100):
  result = expensive_method_that_could_be_outside()
  topic.publish(result + " " + i)
```

If you ran that code, you'd go "wtf, why does this take so long?" and then change `validate` to `False` or -- even better -- move it out of the loop... That's not even Python specific... it's just looping: don't put anything in a loop unless it needs to be there.

That said, considering that unlike boto's `get_bucket()`, no extra information comes back with the topic (ie, the response just sends back "yes here's the name you sent us...") I don't think this is super important.

If topic starts including extra pieces of information (say... a throughput limit of X messages per second?) then it seems like constructing a Topic object on its own means you're missing out on some important information, and `.get_topic` (with `validate=True` by default) seems like a good choice. If/when that debate comes up, we can discuss it (obviously making `.topic` do a lookup down the road would be bad... as everyone who treated it as though it wasn't making an API call will now get shot in the foot).